cmake/templates/oneDPLConfig.cmake.in

```diff
@@ -25,8 +25,10 @@ if (EXISTS "${_onedpl_headers}")
 
         # This compiler check/requirement does not work on Windows with dpcpp and dpcpp-cl compiler drivers
         # icx-cl and icpx are not affected
-        if (NOT (WIN32 AND CMAKE_CXX_COMPILER MATCHES ".*dpcpp(-cl)?(.exe)?$"))
+        if (NOT (WIN32 AND CMAKE_CXX_COMPILER_ID MATCHES "IntelLLVM Workaround"))
```

Comment:
I recently noticed that `cmake/windows-dpcpp-toolchain.cmake` is not a part of oneDPL installed package. For example, oneDPL installed through a [base toolkit](https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html) (online, offline installers) or through a [conda package](https://anaconda.org/intel/onedpl-devel).

We should either not to rely on its inclusion here or to add it.
It is also not straightforward that this file should be passed to cmake. Currently, I do not know if the toolchain file can be picked up automatically by cmake, e.g. basing on the OS and the compiler. We should thing about enabling it or adding some instructions for users.

Reply:
Would you have objections to removing this toolchain file, and rather just incorporating its contents into the main CMakeLists.txt file and just enabling it on windows with `icpx` and `dpcpp`?
I am not convinced that it belongs as a toolchain file exactly, it seems outside of our general pattern for compilers, OS, targets.

Reply:
Sounds good. I have no objections. However, we should duplicate the logic in both `CMakeLists.txt` and `oneDPLConfig.cmake.in`.